library/include/modules/Units.h

```diff
@@ -89,6 +89,9 @@ DFHACK_EXPORT df::general_ref *getGeneralRef(df::unit *unit, df::general_ref_typ
 DFHACK_EXPORT df::specific_ref *getSpecificRef(df::unit *unit, df::specific_ref_type type);
 
 DFHACK_EXPORT df::item *getContainer(df::unit *unit);
+/// what is the outermost object it is contained in? Possible ref types: UNIT, ITEM_GENERAL, VERMIN_EVENT
+DFHACK_EXPORT void getOuterContainerRef(df::specific_ref &spec_ref, df::unit *unit, bool init_ref=true);
+DFHACK_EXPORT inline df::specific_ref getOuterContainerRef(df::unit *unit) { df::specific_ref s; getOuterContainerRef(s, unit); return s; }
```

Comment:
I think you were correct in your earlier assessment in e0f4cacf7b1da3ad2fa12d28c772ee7f7a8e23f9. Pretty much every Lua-exposed function I've found returns a primitive or a pointer, not a struct by value. Custom tables are a decent alternative, like in https://github.com/DFHack/dfhack/pull/1992/commits/91c98d4b5180b3b767d037482538931719712f80